src/routes/balances/__tests__/controllers/zerion-balances.controller.spec.ts

```diff
@@ -425,5 +442,153 @@ describe('Balances Controller (Unit)', () => {
+              fiatTotal: expect.any(String),
+              items: expect.any(Array),
+            });
+          });
+
+        expect(networkService.get.mock.calls.length).toBe(2);
+      });
+
+      it('triggers a rate-limit error', async () => {
+        const chain = chainBuilder().with('chainId', zerionChainIds[0]).build();
+        const safeAddress = faker.finance.ethereumAddress();
+        const chainName = app
+          .get(IConfigurationService)
+          .getOrThrow(
+            `balances.providers.zerion.chains.${chain.chainId}.chainName`,
+          );
+        const nativeCoinFungibleInfo = zerionFungibleInfoBuilder()
+          .with('implementations', [
+            zerionImplementationBuilder()
+              .with('address', null)
+              .with('chain_id', chainName)
+              .build(),
+          ])
+          .build();
+        const erc20TokenFungibleInfo = zerionFungibleInfoBuilder()
+          .with('implementations', [
+            zerionImplementationBuilder().with('chain_id', chainName).build(),
+          ])
+          .build();
+        const zerionApiBalancesResponse = zerionBalancesBuilder()
+          .with('data', [
+            zerionBalanceBuilder()
+              .with(
+                'attributes',
+                zerionAttributesBuilder()
+                  .with('fungible_info', erc20TokenFungibleInfo)
+                  .build(),
+              )
+              .build(),
+            zerionBalanceBuilder()
+              .with(
+                'attributes',
+                zerionAttributesBuilder()
+                  .with('fungible_info', nativeCoinFungibleInfo)
+                  .build(),
+              )
+              .build(),
+          ])
+          .build();
+        networkService.get.mockImplementation((url) => {
+          switch (url) {
+            case `${safeConfigUrl}/api/v1/chains/${chain.chainId}`:
+              return Promise.resolve({ data: chain, status: 200 });
+            case `${zerionBaseUri}/v1/wallets/${safeAddress}/positions`:
+              return Promise.resolve({
+                data: zerionApiBalancesResponse,
+                status: 200,
+              });
+            default:
+              return Promise.reject(new Error(`Could not match ${url}`));
+          }
+        });
+
+        const limitCalls = configurationService.getOrThrow<number>(
+          'balances.providers.zerion.limitCalls',
+        );
+        for (let i = 0; i < limitCalls; i++) {
+          await request(app.getHttpServer())
+            .get(
+              `/v1/chains/${chain.chainId}/safes/${safeAddress}/balances/${crypto.randomUUID()}`,
+            )
+            .expect(200);
+        }
+
+        await request(app.getHttpServer())
+          .get(
+            `/v1/chains/${chain.chainId}/safes/${safeAddress}/balances/${crypto.randomUUID()}`,
+          )
+          .expect(429);
+
+        expect(networkService.get.mock.calls.length).toBe(limitCalls + 1);
```

Comment:
Nit: can we add a comment here regarding the `+ 1` relating to the Config Service call? It initially confused me.

Reply:
Absolutely! Good point.

Added in [395394b](https://github.com/safe-global/safe-client-gateway/pull/1228/commits/395394bf4b4f8fbc44293488ddf74f17c798ce36)